File: social-service/src/main/java/com/eirsteir/coffeewithme/social/exception/ExceptionResponse.java
package com.eirsteir.coffeewithme.social.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.util.Date;

@Getter
public class ExceptionResponse {

    private Date timestamp;
    private String message;
    private String details;
    private String status;

    public ExceptionResponse(Date timestamp, String message, String details, HttpStatus status) {
        super();
        this.timestamp = timestamp;
        this.message = message;
        this.details = details;
        this.status = status.toString();
    }
}

File: commons/src/main/kotlin/com/eirsteir/coffeewithme/commons/exception/ExceptionResponse.kt
package com.eirsteir.coffeewithme.commons.exception

import org.springframework.http.HttpStatus
import java.time.LocalDateTime

data class ExceptionResponse(
    private var timestamp: LocalDateTime = LocalDateTime.now(),
    var status: HttpStatus,
    private var details: String,
    private var message: String? = "Something went wrong",
    private var errors: List<String> = listOf()
)